rust/src/cliwrap/yumdnf.rs

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{anyhow, Result};
+use indoc::formatdoc;
+use indoc::indoc;
+use std::os::unix::process::CommandExt;
+use std::path::Path;
+use std::process::Command;
+use structopt::StructOpt;
+
+/// Emitted at the first line.
+const IMAGEBASED: &str = "Note: This system is image (rpm-ostree) based.";
+
+/// Emitted for unhandled options.
+const UNHANDLED: &str = indoc::indoc! {r#"
+rpm-ostree: Unknown arguments passed to yum/dnf compatibility wrapper.
+
+Please see `yum --help` for currently accepted commands via this
+wrapper, and `rpm-ostree --help` for more information about.the rpm-ostree
+project.
+"#};
+
+const OTHER_OPTIONS: &[(&str, &str)] = &[
+    (
+        "toolbox",
+        "For command-line development and debugging tools in a privileged container",
+    ),
+    ("podman", "General purpose containers"),
+    ("docker", "General purpose containers"),
+    ("flatpak", "Desktop (GUI) applications"),
+];
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "yumdnf-rpmostree",
+    about = "Compatibility wrapper implementing subset of yum/dnf CLI"
+)]
+#[structopt(rename_all = "kebab-case")]
+/// Main options struct
+enum Opt {
+    /// Start an upgrade of the operating system
+    Upgrade,
+    /// Start an upgrade of the operating system
+    Update,
+    /// Display information about system state
+    Status,
+    /// Perform a search of packages.
+    Search {
+        /// Search terms
+        terms: Vec<String>,
+    },
+    /// Will return an error suggesting other approaches.
+    Install {
+        /// Set of packages to install
+        packages: Vec<String>,
+    },
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum RunDisposition {
+    HelpOrVersionDisplayed,
+    ExecRpmOstree(Vec<String>),
+    UseSomethingElse,
+    NotImplementedYet(&'static str),
+    Unhandled,
+}
+
+fn disposition(argv: &[&str]) -> Result<RunDisposition> {
+    let opt = match Opt::from_iter_safe(std::iter::once(&"yum").chain(argv.iter())) {
+        Ok(v) => v,
+        Err(e)
+            if e.kind == clap::ErrorKind::VersionDisplayed
+                || e.kind == clap::ErrorKind::HelpDisplayed =>
+        {
+            return Ok(RunDisposition::HelpOrVersionDisplayed)
+        }
+        Err(_) => {
+            return Ok(RunDisposition::Unhandled);
+        }
+    };
+
+    let disp = match opt {
+        Opt::Upgrade | Opt::Update => RunDisposition::ExecRpmOstree(vec!["upgrade".into()]),
+        Opt::Status => RunDisposition::ExecRpmOstree(vec!["status".into()]),
```

Comment:
Hmm OK, so this then allows users to type `yum status`, which isn't a valid command in `dnf`/`yum`. Mostly OK with this I guess though if the primary goal is yum/dnf CLI compatibility, and `status` is not a valid yum/dnf command, then users would never know to type this and there's not much point for us to support it. It's like we'd be sending mixed signals. Unless where you're going with this is to revive https://github.com/coreos/rpm-ostree/issues/405 and push for `rpm-ostree` to be renamed to `yum`. :)

Reply:
Not rename.  But I think my proposal is to make `yum status` a thing even for traditional yum.  Even if it doesn't do the same thing to start.